sia/src/types/v1.rs

```diff
@@ -883,25 +912,111 @@ mod tests {
     #[test]
     fn test_serialize_transaction() {
         let transaction = Transaction {
-            siacoin_inputs: Vec::new(),
-            siacoin_outputs: Vec::new(),
-            file_contracts: Vec::new(),
-            file_contract_revisions: Vec::new(),
-            storage_proofs: Vec::new(),
+            siacoin_inputs: vec![SiacoinInput {
+                parent_id: SiacoinOutputID::parse_string(
+                    "750d22eff727689d1d8d1c83e513a30bb68ee7f9125a4dafc882459e34c2069d",
+                )
+                .unwrap(),
+                unlock_conditions: UnlockConditions {
+                    timelock: 0,
+                    public_keys: vec![UnlockKey::parse_string(
+                        "ed25519:800ed6c2760e3e4ba1ff00128585c8cf8fed2e3dc1e3da1eb92d49f405bd6360",
+                    )
+                    .unwrap()],
+                    signatures_required: 6312611591377486220,
+                },
+            }],
+            siacoin_outputs: vec![SiacoinOutput {
+                value: Currency::new(890415399000000000000000000000000),
+                address: Address::parse_string(
+                    "480a064b5fca13002a7fe575845154bbf0b3af4cc4f147cbed387d43cce3568ae2497366eaa7",
+                )
+                .unwrap(),
+            }],
+            file_contracts: vec![
+                FileContract{
+                    file_size: 0,
+                    file_merkle_root: Hash256::default(),
+                    window_start: 10536451586783908586,
+                    window_end: 9324702155635244357,
+                    payout: Currency::new(0),
+                    unlock_hash: Address::default(),
+                    revision_number: 9657412421282982780,
+                    valid_proof_outputs: vec![
+                        SiacoinOutput{
+                            value: Currency::new(1933513214000000000000000000000000),
+                            address: Address::parse_string(
+                                "944524fff2c49c401e748db37cfda7569fa6df35b704fe716394f2ac3f40ce87b4506e9906f0",
+                            ).unwrap(),
+                        }
+                    ],
+                    missed_proof_outputs: vec![
+                       SiacoinOutput{
+                            value: Currency::new(2469287901000000000000000000000000),
+                            address: Address::parse_string(
+                                "1df67838262d7109ffcd9018f183b1eb33f05659a274b89ea6b52ff3617d34a770e9dd071d2e",
+                            ).unwrap(),
+                       }
+                    ]
+                }
+            ],
+            file_contract_revisions: vec![
+                FileContractRevision{
+                   parent_id: FileContractID::parse_string(
```

Comment:
Maybe you can extend `ImplHashID` to also declare a macro for each type similar to the `address!` macro. To clean these calls up as well. e.g. `file_contract_id!` or `fileContractID!` (or whatever the convention is). Assuming that is possible.